tests/client/coll.rs

```diff
@@ -1,50 +1,303 @@
+        };
+    }
+}
+
+#[test]
+fn update_one() {
+    let client = MongoClient::with_uri("mongodb://localhost:27017").unwrap();
+    let db = client.db("test");
+    let coll = db.collection("test");
+
+    db.drop_database().ok().expect("Failed to drop database");
+
+    // Insert documents
+    let mut doc = bson::Document::new();
+    let mut doc2 = bson::Document::new();
+    let mut doc3 = bson::Document::new();
+
+    doc.insert("title".to_owned(), Bson::String("Jaws".to_owned()));
+    doc2.insert("title".to_owned(), Bson::String("Back to the Future".to_owned()));
+    doc3.insert("title".to_owned(), Bson::String("12 Angry Men".to_owned()));
+    coll.insert_many(vec!(doc.clone(), doc2.clone(), doc3.clone()), false, None)
+        .ok().expect("Failed to insert documents into collection.");
+    
+    // Update single document
+    let mut update = bson::Document::new();
+    let mut set = bson::Document::new();
+
+    set.insert("director".to_owned(), Bson::String("Robert Zemeckis".to_owned()));
+    update.insert("$set".to_owned(), Bson::Document(set));
+    coll.update_one(doc2.clone(), update, false, None).ok().expect("Failed to update document.");
+
+    let results = coll.find(None, None).ok().expect("Failed to execute find command.");
+    assert_eq!(3, results.len());
+
+    // Assert director attributes
+    assert!(results[0].get("director").is_none());
+    assert!(results[2].get("director").is_none());
+    match results[1].get("director") {
+        Some(&Bson::String(ref director)) => assert_eq!("Robert Zemeckis", director),
+        _ => panic!("Expected Bson::String!"),
+    }
+}
+
+#[test]
+fn update_many() {
+    let client = MongoClient::with_uri("mongodb://localhost:27017").unwrap();
+    let db = client.db("test");
+    let coll = db.collection("test");
+
+    db.drop_database().ok().expect("Failed to drop database");
+
+    // Insert documents
+    let mut doc = bson::Document::new();
+    let mut doc2 = bson::Document::new();
+    let mut doc3 = bson::Document::new();
+
+    doc.insert("title".to_owned(), Bson::String("Jaws".to_owned()));
+    doc2.insert("title".to_owned(), Bson::String("Back to the Future".to_owned()));
+    doc3.insert("title".to_owned(), Bson::String("12 Angry Men".to_owned()));
+    coll.insert_many(vec!(doc.clone(), doc2.clone(), doc3.clone(), doc2.clone()), false, None)
+        .ok().expect("Failed to insert documents into collection.");
+    
+    // Update single document
+    let mut update = bson::Document::new();
+    let mut set = bson::Document::new();
+
+    set.insert("director".to_owned(), Bson::String("Robert Zemeckis".to_owned()));
+    update.insert("$set".to_owned(), Bson::Document(set));
+    coll.update_many(doc2.clone(), update, false, None).ok().expect("Failed to update documents.");
+
+    let results = coll.find(None, None).ok().expect("Failed to execute find command.");
+    assert_eq!(4, results.len());
+
+    // Assert director attributes
+    for i in 0..3 {
```

Comment:
Yep, working on it now and there's a bug in the `update_many` test, will open a new PR.